evm/src/cpu/kernel/asm/core/terminate.asm

```diff
@@ -103,7 +159,7 @@ global fault_exception:
     // stack: (empty)
     PUSH 0 // leftover_gas
     // TODO: Revert state changes.
-    // TODO: Set parent context's CTX_METADATA_RETURNDATA_SIZE to 0.
+    %set_parent_returndata_size_zero
     PUSH 0 // success
     %jump(terminate_common)
```

Comment:
I'm not sure I understand why we don't panic here. Seems like going to `terminate_common` will just continue the execution?

Reply:
My understanding is that for exceptions of this sort, the block is still valid (so we still need to prove the result), and the txn could potentially still continue if there's a parent context (besides the kernel).